plugins/modules/smart_proxy.py

```diff
@@ -101,13 +110,53 @@ def main():
         foreman_spec=dict(
             name=dict(required=True),
             url=dict(required=True),
+            lifecycle_environments=dict(required=False, type='entity_list'),
             download_policy=dict(required=False, choices=['background', 'immediate', 'on_demand']),
         ),
-        required_plugins=[('katello', ['download_policy'])],
+        required_plugins=[('katello', ['lifecycle_environments', 'download_policy'])],
     )
 
     with module.api_connection():
-        module.run()
+        handle_lifecycle_environments = not module.desired_absent and 'lifecycle_environments' in module.foreman_params
+        if handle_lifecycle_environments:
+            module.lookup_entity('lifecycle_environments')
+
+        lifecycle_environments = module.foreman_params.pop('lifecycle_environments', None)
+        smart_proxy = module.run()
+
+        payload = {
+            'id': smart_proxy['id'],
+        }
+        entity = module.resource_action('capsule_content', 'lifecycle_environments', payload, record_change=False)
+
+        if handle_lifecycle_environments:
+            desired_environment_ids = set(lifecycle_environment['id'] for lifecycle_environment in lifecycle_environments)
+            current_environment_ids = set(lifecycle_environment['id'] for lifecycle_environment in entity['results']) if entity else set()
+
+            module.record_before('smart_proxy_content/lifecycle_environment_ids', current_environment_ids)
+            module.record_after('smart_proxy_content/lifecycle_environment_ids', desired_environment_ids)
+            module.record_after_full('smart_proxy_content/lifecycle_environment_ids', desired_environment_ids)
+
+            if desired_environment_ids != current_environment_ids:
+                if not module.check_mode:
```

Comment:
Is that needed? `resource_action` is noop if we're in check mode (and didn't pass `ignore_check_mode`) and will call `set_changed` for us.

Reply:
I removed it and it still works as I want, so I guess not. This would warrant a closer look at the code from the sync_plan module. As I copied that code to make this.